tool-openssl/crl.cc

```diff
@@ -0,0 +1,87 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+#include "internal.h"
+#include <openssl/x509.h>
+#include <openssl/pem.h>
+#include <ctime>
+
+static const argument_t kArguments[] = {
+        { "-help", kBooleanArgument, "Display option summary" },
+        { "-in", kOptionalArgument, "Input file, default stdin" },
+        { "-hash", kBooleanArgument, "Print hash value" },
+        { "-fingerprint", kBooleanArgument, "Print CRL fingerprint" },
+        { "-noout", kBooleanArgument, "No CRL output" },
+        { "", kOptionalArgument, "" }
+};
+
+bool CRLTool(const args_list_t &args) {
+  args_map_t parsed_args;
+  if (!ParseKeyValueArguments(&parsed_args, args, kArguments)) {
+    PrintUsage(kArguments);
+    return false;
+  }
+
+  std::string in;
+  bool help = false, hash = false, fingerprint = false, noout = false;
+
+  GetBoolArgument(&help, "-help", parsed_args);
+  GetBoolArgument(&hash, "-hash", parsed_args);
+  GetBoolArgument(&fingerprint, "-fingerprint", parsed_args);
+  GetBoolArgument(&noout, "-noout", parsed_args);
+  GetString(&in, "-in", "", parsed_args);
```

Comment:
NP: It would be helpful for the order here to be the same as above:
```
  GetBoolArgument(&help, "-help", parsed_args);
  GetString(&in, "-in", "", parsed_args);
  GetBoolArgument(&hash, "-hash", parsed_args);
  GetBoolArgument(&fingerprint, "-fingerprint", parsed_args);
  GetBoolArgument(&noout, "-noout", parsed_args);
```